packages/dashboard-core-plugins/src/panels/NotebookPanel.tsx

```diff
@@ -600,6 +607,41 @@ class NotebookPanel extends Component<NotebookPanelProps, NotebookPanelState> {
     this.setState({ showCloseModal: false });
   }
 
+  /**
+   * Close overwritten tabs
+   * @params {string} fileName = The name of the file to be overwritten
+   */
+  handleOverwrite(fileName: string): void {
+    const { glContainer, glEventHub } = this.props;
+    const { fileMetadata } = this.state;
+    const tabs = glContainer.tab?.header.tabs;
+    const tabsToBeRemoved = tabs?.filter(
+      (tab: Tab) =>
+        tab.contentItem.config.title === fileName && tab.isActive === false
+    );
+
+    // Close old overwritten tabs
+    if (tabsToBeRemoved && tabsToBeRemoved.length > 0) {
+      for (let i = 0; i < tabsToBeRemoved?.length; i += 1) {
+        if (isComponent(tabsToBeRemoved[i].contentItem)) {
+          (tabsToBeRemoved[i].contentItem as glComponent).container.close();
+        } else {
+          tabsToBeRemoved[i].header.parent.removeChild(
+            tabsToBeRemoved[i].contentItem
+          );
+        }
+      }
+    }
```

Comment:
This is the wrong way to go about it:
- It's only looking in the same tab stack as the current tab
- It ends up closing the tab that you just renamed as well.

I think instead what you should be doing is emitting the `NotebookEvent.RENAME_FILE` event, and then `NotebookPanel` can listen for that, and if it picks up that the new file name is matching itself and it's not the panel we renamed from, then it can close itself.